arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,150 @@
+        }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        // Present the controller.
+        alertController.popoverPresentationController?.barButtonItem = queryBarButtonItem
+        present(alertController, animated: true)
+    }
+    
+    /// The service request URL.
+    static let serviceRequestURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/ServiceRequest/MapServer")!
+    /// The map image layer that uses the service URL.
+    let serviceRequestsMapImageLayer = AGSArcGISMapImageLayer(url: serviceRequestURL)
+    /// The array to store the possible comments.
+    var commentsArray = [AGSFeature]()
+    /// The graphics overlay to add graphics to.
+    let selectedFeaturesOverlay = AGSGraphicsOverlay()
+    
+    func makeMap() -> AGSMap {
+        // Create a map with the ArcGIS streets basemap style.
+        let map = AGSMap(basemapStyle: .arcGISStreets)
+        // Load the map image layer's tables and layers.
+        serviceRequestsMapImageLayer.loadTablesAndLayers { _ in
+            // Add the layer to the map.
+            map.operationalLayers.add(self.serviceRequestsMapImageLayer)
+            self.queryFeatures()
+        }
+        return map
+    }
+    
+    /// Query features on the first table in the map image layer.
+    func queryFeatures() {
+        // Create query parameters and set its where clause.
+        let nullCommentsParameters = AGSQueryParameters()
+        nullCommentsParameters.whereClause = "requestid <> '' AND comments <> ''"
+        // Get the first table from the map image layer.
+        guard let commentsTable = serviceRequestsMapImageLayer.tables.first else { return }
+        // Query features on the feature table with the query parameters and all feature fields.
+        commentsTable.queryFeatures(with: nullCommentsParameters, queryFeatureFields: .loadAll) { [weak self] result, error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Show the records from the service request comments table in the list view control.
+                self.commentsArray = (result?.featureEnumerator().allObjects)!
+                // Enable the button after the map and features have been loaded.
+                self.queryBarButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    /// Query related features for the selected feature.
+    func queryCommentsTable(feature: AGSFeature, commentsTable: AGSServiceFeatureTable, relatedQueryParameters: AGSRelatedQueryParameters) {
+        // Get the feature as an ArcGIS feature.
+        guard let selectedComment = feature as? AGSArcGISFeature else { return }
+        // Query related features for the selected comment and its related query parameters.
```

Comment:
Same as comment above, maybe consider constructing the query parameter from the commentsTable in this method.